nixos/tests/vikunja.nix

```diff
@@ -0,0 +1,66 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "vikunja";
+
+  meta = with lib.maintainers; {
+    maintainers = [ em0lar ];
+  };
+
+  nodes = {
+    vikunjaSqlite = { ... }: {
+      services.vikunja = {
+        enable = true;
+        database = {
+          type = "sqlite";
+        };
+        frontendScheme = "http";
+        frontendHostname = "localhost";
+      };
+      services.nginx.enable = true;
+    };
+    vikunjaPostgresql = { pkgs, ... }: {
+      services.vikunja = {
+        enable = true;
+        database = {
+          type = "postgres";
+          user = "vikunja";
+          database = "vikunjadb";
+        };
+        frontendScheme = "http";
+        frontendHostname = "localhost";
+        environmentFiles = [
+          (pkgs.writeText "vikunja-env" ''
+            VIKUNJA_DATABASE_PASSWORD=snakeoilpwd
+          '')
+        ];
+      };
+      services.postgresql = {
+        enable = true;
+        initialScript = pkgs.writeText "pg-init-script.sql" ''
+         CREATE ROLE vikunja LOGIN PASSWORD 'snakeoilpwd';
+         CREATE DATABASE vikunjadb OWNER vikunja;
+        '';
+      };
+      services.nginx.enable = true;
+    };
+  };
+
+  testScript =
+    ''
+      vikunjaSqlite.wait_for_unit("vikunja-api.service")
+      vikunjaSqlite.wait_for_open_port(3456)
+      vikunjaSqlite.succeed("curl --fail http://localhost:3456/api/v1/info")
+
+      vikunjaSqlite.wait_for_unit("nginx.service")
+      vikunjaSqlite.wait_for_open_port(80)
+      vikunjaSqlite.succeed("curl --fail http://localhost/api/v1/info")
```

Comment:
Might want to add another check for the frontend as well to see if it is served correctly, not only if the nginx proxies the api requests correctly.

If we want to go really fancy we could also compare the outputs of `http://localhost:3456/api/v1/info` and `http://localhost/api/v1/info` but I'm not sure if that has any benefit.

Reply:
I added a check for the frontend part. The other one I would save for the future